clusterloader2/pkg/measurement/common/generic_query_measurement.go

```diff
@@ -43,108 +44,103 @@ func init() {
 }
 
 type genericQueryGatherer struct {
-	metricName    string
-	metricVersion string
-	queries       []genericQuery
-	unit          string
+	StartParams
 }
 
-type genericQuery struct {
-	name         string
-	query        string
-	hasThreshold bool
-	threshold    float64
+// StartParams represents configuration that can be passed as params
+// with action: start.
+type StartParams struct {
+	MetricName    string
+	MetricVersion string
+	Queries       []GenericQuery
+	Unit          string
 }
 
-func (g *genericQueryGatherer) Configure(config *measurement.Config) error {
-	metricName, err := util.GetString(config.Params, "metricName")
-	if err != nil {
-		return err
+// TODO(mborsz): github.com/go-playground/validator or similar project?
+func (p *StartParams) Validate() error {
+	if p.MetricName == "" {
+		return goerrors.New("metricName is required")
 	}
-
-	metricVersion, err := util.GetString(config.Params, "metricVersion")
-	if err != nil {
-		return err
+	if p.MetricVersion == "" {
+		return goerrors.New("metricVersion is required")
 	}
-
-	unit, err := util.GetString(config.Params, "unit")
-	if err != nil {
-		return err
+	if p.Unit == "" {
+		return goerrors.New("unit is required")
 	}
 
-	queries, err := util.GetMapArray(config.Params, "queries")
-	if err != nil {
-		return err
-	}
-	var genericQueries []genericQuery
-	for _, q := range queries {
-		query, err := util.GetString(q, "query")
-		if err != nil {
-			return err
-		}
-		name, err := util.GetString(q, "name")
-		if err != nil {
-			return err
-		}
-		hasThreshold := true
-		threshold, err := util.GetFloat64(q, "threshold")
-		if util.IsErrKeyNotFound(err) {
-			klog.V(2).Infof("No threshold set for %v: %v", metricName, name)
-			hasThreshold = false
-		} else if err != nil {
-			return err
+	for idx, query := range p.Queries {
+		if err := query.Validate(); err != nil {
+			return fmt.Errorf("params.queries[%d] validation failed: %v", idx, err)
 		}
-		genericQueries = append(genericQueries, genericQuery{name, query, hasThreshold, threshold})
 	}
 
-	g.metricName = metricName
-	g.metricVersion = metricVersion
-	g.unit = unit
-	g.queries = genericQueries
 	return nil
 }
 
+type GenericQuery struct {
+	Name      string
+	Query     string
+	Threshold *float64
+}
+
+func (q *GenericQuery) Validate() error {
+	if q.Name == "" {
+		return goerrors.New("name is required")
+	}
+	if q.Query == "" {
+		return goerrors.New("query is required")
```

Comment:
I'm leaning towards switching to github.com/go-playground/validator in the next PR, so let's not invest too much in this logic as it will be likely removed.